ctapipe/io/hdf5eventsource.py

```diff
@@ -60,6 +63,7 @@
 
 COMPATIBLE_DATA_MODEL_VERSIONS = [
     "v4.0.0",
+    "v4.1.0",
```

Comment:
Since `4.0.0` was never part of a released version of ctapipe, I'd be inclined to not make it a new version and to just include it in version `4.0.0`. In the other PRs changing the date model I only updated the changelog for 4.0, not bumped the version each time.

Reply:
There should be no reason for us to support reading files without the obsconfig / sb config tables now.

Reply:
ok, I'll unbump it (just wasnt sure if any files were written already with the dev version)